microsoft/PowerToys
# [PTRun] Implement thread-safety in query results task with lock

## Summary of the Pull Request
When PTRun first opened, it throws an exception in QueryResults function at first query. This error is solved by using _updateSource.Token instead of local currentCancellationToken to avoid dangling reference.


## PR Checklist

- [ ] **Closes:** #30848 


## Detailed Description of the Pull Request / Additional comments


## Validation Steps Performed

- Disable PTRun and enable it again
- Copy "%% 10 ft in m"
- Run PTRun with hotkey and paste "%% 10 ft in m"
- Observe the result is shown or not
